QuestionAnswering/QA_ICL/data_generation/README.md

```diff
@@ -1,90 +1,155 @@
+  - [Installation](#installation)
+    - [Prerequisites](#prerequisites)
+    - [Steps](#steps)
+  - [Usage](#usage)
+    - [Lexicon Generation](#lexicon-generation)
+    - [KG Schema Extraction](#kg-schema-extraction)
+    - [CSV-to-JSON Conversion of Data Request Generation Examples](#csv-to-json-conversion-of-data-request-generation-examples)
+    - [Augmentation of Data Request Generation](#augmentation-of-data-request-generation)
+    - [CSV-to-JSON Conversion of Physical Quantity Detection Examples](#csv-to-json-conversion-of-physical-quantity-detection-examples)
+  - [JSON Schema Definitions](#json-schema-definitions)
+    - [JSON Schema of Lexicons](#json-schema-of-lexicons)
+    - [JSON schema of KG schema elements](#json-schema-of-kg-schema-elements)
+    - [JSON Schema of Data Request Generation Examples](#json-schema-of-data-request-generation-examples)
+    - [JSON Schema of Physical Quantity Recognition Examples](#json-schema-of-physical-quantity-recognition-examples)
+  
+## Overview
+
+This directory contains Python and shell scripts to prepare three kinds of data required by Marie's backend: lexicons, graph schemas, and examples of data request generation.
+
+- Lexicons: help with entity linking
+- Graph schemas: help with predicting SPARQL queries that look for information needed by users
+- Examples of data request generation: provide demonstrations for how user queries ought to be mapped to structured data requests.
+- Examples of physical quantity recognition: provide demonstrations for how to extract mentions of physical quantities from input texts.
+ 
+Specifically, the scripts in this directory provide the following functionalities:
+
+- Generate lexicons from ABoxes or TBoxes for all instances of an particular `rdf:type`.
+
+- Extract schema elements from ABoxes or TBoxes.
+
+- Convert the format of labelled examples, including examples for data request generation and physical quantity recognition, from CSV to JSON.
+  
+  When labelled examples are hand-crafted by human developers, it is more convenient to do so in a spreadsheet than crafting a JSON file from scratch. The spreadsheet can then be exported into the CSV format and converted to JSON so that it can be loaded into the backend application.
+
+- Augment data request generation examples with synthetic entries.
+
+  One of the factors that influence the success of in-context learning with demonstration retrieval is the diversity of in-context demonstrations. Using an LLM, an initially small set of examples can be augmented to enhance its size and diversity.
+
+
+## Terminologies
 
 ### Lexicon
 
 Within the context of this project, a lexicon of a class, a predicate, or an entity is defined as a collection of its text representations, which include its canonical label and surface forms. 
 
-In the main application, lexicons are used to perform:
+See [JSON schema for lexicons](#json-schema-of-lexicons).
```

Comment:
Put the JSON schema right here, so that it flows better. Clicking on the link breaks my chain of thought.

Also explain if the JSON schema is only applicable to this specific QA system or not.